packages/jaeger-ui/src/utils/readJsonFile.tsx

```diff
@@ -23,20 +23,43 @@ export default function readJsonFile(fileList: { file: File }): Promise<string>
         return;
       }
       try {
-        const traceObj = JSON.parse(reader.result);
-        if ('resourceSpans' in traceObj) {
-          JaegerAPI.transformOTLP(traceObj)
-            .then((result: string) => {
-              resolve(result);
-            })
-            .catch(() => {
-              reject(new Error(`Error converting traces to OTLP`));
-            });
-        } else {
-          resolve(traceObj);
+        const resourceSpansArray: any[] = [];
```

Comment:
this solution is very complicated. Please see the other PR https://github.com/jaegertracing/jaeger-ui/pull/2254/files#diff-3c0ff4ee517b43a5f908b13a32c87f9bc9f63244442562274f6ae7707511a92a which already had (I think) the right solution, but was missing unit tests.

Reply:
I can do changes according to that PR, are we sure to discard this approach?

Reply:
Although complicated code wise, but I feel the approach is pretty simple

- Check whether JSON.parse works, if it doesn't, move to the catch block where it pushes everything into an array and treats it as a normal structure. I do feel writing so much business logic in a try-catch might not be the go to, but yeah, open for suggestions.

Reply:
please look at the other PR first. That change is 6-7 lines to the main logic. Your is about 40 lines of pretty convoluted logic, with several branches, some of which are only invoked via exception handling. You are not going to convince me that yours is better. Simpler is always better.

Reply:
Checked the PR. I have one question there:

```
try {
        let traceObj = JSON.parse(reader.result);
        if (Array.isArray(traceObj) && traceObj.every(obj => 'resourceSpans' in obj)) {
          const mergedResourceSpans = traceObj.reduce((acc, obj) => {
            acc.push(...obj.resourceSpans);
            return acc;
          }, []);

          traceObj = { resourceSpans: mergedResourceSpans };
}
```


The `let traceObj = JSON.parse(reader.result);` should fail immediately and go to the catch block upon reading a JSON file like the one given in the issue, so it goes on to execute L47 in that code, which is throwing the error. 

Am I missing something obvious, because sometimes the smallest detail makes the difference and I might have missed it.

Reply:
ah, so you are assuming that the overall input is not going to be a valid JSON? That's a valid point, we _could_ use try/catch to check for that. But still, in this case I would structure the code like this:
```
try {
  traceObj = JSON.parse()
} catch() {
  // try to interpret input as JSON-per-line
  try {
    traceObj = tryParseMultiLineInput(reader.result) // call helper function
  } catch() {
    // log error
    return
  }
}
// now that we have a valid JSON object, follow the logic from the other PR.
// preferably
// 1. check if array and transform to valid OTLP
// 2. check if OTLP and transform to Jaeger model
// 3. parse Jaeger model
```

Reply:
Right, I'll implement it this way.

Reply:
Hi, @yurishkuro I've updated the PR. However, yarn prettier would fail because the json test file has a wrong json format (due to being seperated by newlines). Can we ignore this file in the config? What is the way to go about this.

Reply:
perhaps you can rename the file to e.g. `*.json.txt`